legion/engine/physics/physx/systems/physx_physics_system.cpp

```diff
@@ -166,12 +166,12 @@ namespace legion::physics
         for (auto entity : physicsAndRigidbodyComponentFilter)
         {
             auto& physComp = *entity.get_component<physics_component>();
-            auto optionalPhysxWrapper = m_physxWrapperContainer.findWrapperWithID(physComp.physicsComponentID);
+            auto physxWrapper = m_physxWrapperContainer.findWrapperWithID(physComp.physicsComponentID).value();
 
-            if (optionalPhysxWrapper->get().bodyType != physics::physics_body_type::rigidbody)
+            if (physxWrapper->bodyType != physics::physics_body_type::rigidbody)
             {
                 //the user has switched this body from static to dynamic
-                optionalPhysxWrapper->get().bodyType = physics::physics_body_type::rigidbody;
+                physxWrapper->bodyType = physics::physics_body_type::rigidbody;
```

Comment:
ig with the way it was you already didn't check whether the returned value was valid? but the function should prolly just return a pointer, and then have a nullptr check here

Reply:
in this specific scenario, findWrapperWithID should always return a value since all invalid physx wrapers have been removed, but I guess a nullptr check wouldn't hurt